plugin/src/test/java/org/opensearch/ml/rest/RestMLInferenceIngestProcessorIT.java

```diff
@@ -390,4 +502,24 @@ protected Map getDocument(final String index, final String docId) throws Excepti
         return parseResponseToMap(docResponse);
     }
 
+    protected MLRegisterModelInput registerModelInput() {
+        MLModelConfig modelConfig = TextEmbeddingModelConfig
+            .builder()
+            .modelType("bert")
+            .frameworkType(TextEmbeddingModelConfig.FrameworkType.SENTENCE_TRANSFORMERS)
```

Comment:
is the local model only supporting for sentence transformers? have you test other type of local models?

Reply:
It is also working on sparse_encoders and cross encoding models. Will add more unit tests

Reply:
Found it difficult to add ITs for other models. Added them in UTs. Currently we dont have predict IT tests for other models. Its not letting me use pre-trained models because of size and timing-out issue. I can add them by first adding a few small model URLs to the test data first. I added a TODO within the test class